### PR TITLE
Changed the the `entity_type` kwarg to `table_name` #111 and `register_entity` now throws exception if user hasn't passed `table_name` #91

### PR DESCRIPTION
**Pull request by kousikmitra**
> #111 
1. Changed the the `entity_type` kwarg to `table_name` 
2. Added Deprecation warning for the `entity_type` kwargs
> #91 
3. Added test function to test whether `register_entity` throws an exception on not passing the `table_name` key
2. Added code to throw `KeyError` exception in `register_entity` function of `FlaskJWTRouter` class if user doesn't pass a `table_name`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+**Unreleased**
```
Comment (joegasewicz) [inline CHANGELOG.md]: Fantastic

File: flask_jwt_router/_jwt_router.py [hunk 2]
```diff
@@ -102,8 +103,14 @@ def register_entity(self, **kwargs) -> str:
         :param kwargs:
         :return: str
         """
-        entity_type = kwargs.get("entity_type")
-        self.extensions.entity_key = self.entity.get_attr_name(entity_type)
+        if 'entity_type' in kwargs:
+            warn(("'entity_type' argument name has been deprecated and will be replaced"
+                  "in the next release. Use 'table_name' instead"))
+            kwargs['table_name'] = kwargs['entity_type']
```
Comment (joegasewicz) [inline flask_jwt_router/_jwt_router.py]: nice

File: tests/test_jwt_router.py [hunk 3]
```diff
@@ -54,7 +54,15 @@ class App:
...
+            def before_request(self, t):
+                pass
+        app = App()
+        flask_jwt_router = JwtRoutes(app)
 
-
-
-
+        with pytest.raises(KeyError, match=r"register_entity.+") as excinfo:
+            token = flask_jwt_router.register_entity(entity_id=1)
```
Comment (joegasewicz) [inline tests/test_jwt_router.py]: great

**Comment by joegasewicz** [comment]
Nice work @kousikmitra, thank you.

**Comment by joegasewicz** [comment]
The API is now looking much better already, thanks

**Comment by kousikmitra** [comment]
@joegasewicz Thanks. Can you please take a look what is the issue with Codacy?

**Comment by joegasewicz** [comment]
> @joegasewicz Thanks. Can you please take a look what is the issue with Codacy?

Screenshot 2019-12-06 at 12 05 32

**Comment by joegasewicz** [comment]
@kousikmitra can you run `make html` to update the docs, also can you update the README.md contributors instructions with this info also, thank you.

**Comment by kousikmitra** [comment]
> 
> 
> @kousikmitra can you run `make html` to update the docs, also can you update the README.md contributors instructions with this info also, thank you.

`make html` is giving this error.(I am running on windows)

    process_begin: CreateProcess(NULL, sphinx-build -M html source build, ...) failed.
    make (e=2): The system cannot find the file specified.
    make: *** [Makefile:20: html] Error 2

**Comment by kousikmitra** [comment]
@joegasewicz why is it showing `get_entity_from_ext` as invalid syntax?

**Comment by joegasewicz** [comment]
> > @kousikmitra can you run `make html` to update the docs, also can you update the README.md contributors instructions with this info also, thank you.
> 
> `make html` is giving this error.(I am running on windows)
> 
> ```
> process_begin: CreateProcess(NULL, sphinx-build -M html source build, ...) failed.
> make (e=2): The system cannot find the file specified.
> make: *** [Makefile:20: html] Error 2
> ```
> 
> Do I need any extra modules for this?

can you install this Sphinx & ill update the Pipfile in my PR, thanks

**Comment by joegasewicz** [comment]
> @joegasewicz why is it showing `get_entity_from_ext` as invalid syntax?

I have added your email to our  codacy organization.

**Comment by kousikmitra** [comment]
> 
> 
> > @joegasewicz why is it showing `get_entity_from_ext` as invalid syntax?
> 
> I have added your email to our codacy organization.

How can I access this?

**Comment by joegasewicz** [comment]
> > > @joegasewicz why is it showing `get_entity_from_ext` as invalid syntax?
> > 
> > 
> > I have added your email to our codacy organization.
> 
> How can I access this?

https://app.codacy.com/manual/joegasewicz/Flask-JWT-Router/dashboard?bid=15313481

**Comment by joegasewicz** [comment]
Jusat this one now to fix, i ignored the inline docs on the abstract methods

Screenshot 2019-12-06 at 13 07 16

**Comment by joegasewicz** [comment]
@kousikmitra let mew know when you have had enough and ill merge your PR in lol

**Comment by joegasewicz** [comment]
Sorry meant to comment and closed by mistake.

**Comment by joegasewicz** [comment]
@kousikmitra let me know when you're happy with the PR & ill merge in, thanks

**Comment by kousikmitra** [comment]
> 
> 
> @kousikmitra let me know when you're happy with the PR & ill merge in, thanks

it's was checking fora `tab` and I was giving a `space` 💢  What to include in the README file?

**Comment by joegasewicz** [comment]
> > @kousikmitra let me know when you're happy with the PR & ill merge in, thanks
> 
> it's was checking fora `tab` and I was giving a `space` 💢 What to include in the README file?

Please add:

To update docs run:
```
make html
```

**Comment by joegasewicz** [comment]
Thanks @kousikmitra, great work!

**Pull request merged**